src/WebJobs.Script/Eventing/File/FileEvent.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Azure.WebJobs.Script.Eventing
+{
+    public sealed class FileEvent : ScriptEvent
+    {
+        public FileEvent(string source, FileSystemEventArgs args) : base(nameof(FileEvent))
+        {
+            Source = source;
+            FileChangeArguments = args;
+        }
+
+        public string Source { get; set; }
```

Comment:
This was meant to identify the source of the `FileEvent`, but I can see your point (and better understand your previous comment). The `ScriptEvent` should really only contain members that are common and required across events, but I agree that this would be a good candidate.